src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -265,7 +311,7 @@ redistributeByKdTree( vtkUnstructuredGrid & mesh )
   rdsf->SetInputDataObject( &mesh );
   rdsf->SetNumberOfPartitions( MpiWrapper::commSize() );
   rdsf->Update();
-  return vtkUnstructuredGrid::SafeDownCast( rdsf->GetOutputDataObject( 0 ) );
+  return vtkDataSet::SafeDownCast( rdsf->GetOutputDataObject( 0 ) );
```

Comment:
FYI (though you are probably already aware of this) `vtkRedistributeDataSetFilter` always outputs a `vtkUnstructuredGrid`, regardless of the type of input dataset. Moreover, the following ParMETIS-based redistribution step is only suitable for unstructured grids. It'll still work here because of the aforementioned conversion, but the results will likely be suboptimal for structured inputs. At the very least I suggest setting `partitionRefinement="0"` in your input files to disable the ParMETIS step. 

Further down the road I would propose avoiding `vtkRedistributeDataSetFilter` altogether, implementing a different partitioning strategy (maybe based on `-x`/`-y`/`-z` flags similar to `InternalMeshGenerator`) and preserving the dataset as `vtkStructuredGrid` until the very end when we have to populate the GEOSX cell blocks. This would probably speed up the loading and reduce memory footprint even further.